packages/web3-utils/src/converters.ts

```diff
@@ -86,7 +86,8 @@ export const bytesToBuffer = (data: Bytes): Buffer | never => {
 	}
 
 	if (typeof data === 'string' && isHexStrict(data)) {
-		return Buffer.from(data.slice(2), 'hex');
+		const finalData = `${/^(..)*$/.test(data) ? '' : '0'}${data.replace(/0(x|X)/, '')}`;
```

Comment:
Mixing up the string interpolation with the regular expressions does make it difficult to read. What do you think about this approach? 

```suggestion
        const eventLength = data.length + (data.length % 2);
		const finalData = data.toLowerCase().replace('0x', '').padStart(eventLength, '0')
```

Reply:
Of course, I'm ok with change! I made it, I just made a modification, because `eventLength` included the `0x` prefix and finally would pad a string with '000`  or  `00`,

Reply:
Sure, please make the change accordingly.